scipy/stats/stats.py

```diff
@@ -6087,11 +6129,11 @@ def ttest_ind(a, b, axis=0, equal_var=True, nan_policy='propagate',
         nan_policy = 'omit'
 
     if contains_nan and nan_policy == 'omit':
-        if permutations or alternative != 'two-sided':
+        if permutations or alternative != 'two-sided' or trim != 0:
```

Comment:
Would you try performing the permutation test on the trimmed statistic?  I think in the `_permutation_ttest` function you just replace `_calc_t_stat` with whatever calculates the trimmed t statistic.